docs/src/test/java/sample/javadsl/StreamWrapperActor.java

```diff
@@ -0,0 +1,80 @@
+package sample.javadsl;
+
+import akka.Done;
+import akka.actor.ActorSystem;
+import akka.actor.UntypedActor;
+import akka.actor.Props;
+import akka.actor.PoisonPill;
+import akka.kafka.ConsumerSettings;
+import akka.kafka.Subscriptions;
+import akka.kafka.javadsl.Consumer;
+import akka.stream.ActorMaterializer;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Sink;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import akka.pattern.Backoff;
+import akka.pattern.BackoffSupervisor;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import scala.concurrent.duration.Duration;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+
+public class StreamWrapperActor extends UntypedActor {
+
+    final ActorSystem system = ActorSystem.create("example");
+
+    protected final Materializer materializer = ActorMaterializer.create(system);
```

Comment:
this can be `ActorMaterializer.create(getContext())`, which will make the materializer a child of the actor, i.e. have the same lifecycle, otherwise you should stop the materializer in the Actor's `postStop`.